src/main/java/edu/harvard/iq/dataverse/util/MailSessionProducer.java

```diff
@@ -0,0 +1,132 @@
+package edu.harvard.iq.dataverse.util;
+
+import edu.harvard.iq.dataverse.settings.JvmSettings;
+import jakarta.annotation.Resource;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Named;
+import jakarta.mail.Authenticator;
+import jakarta.mail.PasswordAuthentication;
+import jakarta.mail.Session;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import java.util.List;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@ApplicationScoped
+public class MailSessionProducer {
+    
+    // NOTE: We do not allow "from" here, as we want the transport to get it from the message being sent, enabling
+    //       matching addresses. If "from" in transport and "from" in the message differ, some MTAs may reject or
+    //       classify as spam.
+    // NOTE: Complete list including descriptions at https://eclipse-ee4j.github.io/angus-mail/docs/api/org.eclipse.angus.mail/org/eclipse/angus/mail/smtp/package-summary.html
+    static final List<String> smtpStringProps = List.of(
```

Comment:
Are those hard-coded lists of some SMTP properties needed?

Reply:
Well, it's for backward-compatibility, right? That's my understanding.

Reply:
Yes and yes. Some folks might want to configure specials they currently have in place with the old way of setting up things. That is the yes to backward compatibility part.

But also in new setups, people might want to be able to use the full extent of Jakarta Mail options. Who are we to tell, what might be necessary for someone in their mail setup? I already removed some options. It could be free form text obviously, but that would mean we loose the ability to check for consistency. I'm not sure what the best way is, so I just went with "lets add them all" for now. If you feel so moved that these lists should be shorter and we provide an option "to set whatever key you want", let me know.